test/test_autodiff_8.cpp

```diff
@@ -8,6 +8,8 @@
 
 BOOST_AUTO_TEST_SUITE(test_autodiff_8)
 
+#if __clang_major__ > 5 || __GNUC__ > 5 || defined(_MSC_VER)
```

Comment:
Should the `#if` block surround the entire test suite? Otherwise as you may have seen:
```
testing.capture-output ../../../bin.v2/libs/math/test/test_autodiff_8.test/clang-linux-5.0.0/release/link-static/threading-multi/visibility-hidden/test_autodiff_8.run
====== BEGIN OUTPUT ======
Test setup error: test tree is empty

EXIT STATUS: 200
====== END OUTPUT ======
```

Reply:
Surrounding the entire test suite yields the same error. I moved the `#if` block to right above the test that fails.